homeassistant/components/http/ban.py

```diff
@@ -118,7 +119,14 @@ async def process_wrong_login(request: Request) -> None:
 
     # The user-agent is unsanitized input so we only include it in the log
     user_agent = request.headers.get("user-agent")
-    log_msg = f"{base_msg} ({user_agent})"
+
+    username_str = ""
+    with suppress(JSONDecodeError):
+        request_json = await request.json()
+        if "username" in request_json:
```

Comment:
`process_wrong_login` is called from any URL that is returning Unauthorized. It feels weird to just print data from random requests.

Reply:
Hey @balloob, thanks for taking a look!
(Silly me forgot to actually post this comment, so it got stuck in _Pending_ for waaay to o long...)

I agree that always printing the username here whenever an auth failure occurs would be weird, as most requests handled will not contain this field. For those requests the field will not be printed at all. Just for the paths in the actual login flow the username field will be present in this log entry.

The main reason I kept this here is that ideally all data for a single failed auth event should be contained in a single log entry. For parsing logs for failed authorization attempts using e.g. CrowdSec it would help to distinguish between retried logins on the same account or random attempts at generic default accounts.

If you have a suggestion to get this data logged by a better suited place in the code base I'm open for suggestions.